pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DownloadParticipantListRoute.java

```diff
@@ -4,91 +4,115 @@
-            Alias alias = Alias.of(column);
-            columnAliasEsPathMap.computeIfAbsent(alias, paths -> new ArrayList<>()).add(column);
-        });
+    /**
+     * Fetches participant information from ElasticSearch in batches of MAX_RESULT_SIZE
+     */
+    private List<ParticipantWrapperDto> fetchParticipantEsData(Filterable filter, QueryParamsMap queryParamsMap) {
+        List<ParticipantWrapperDto> allResults = new ArrayList<ParticipantWrapperDto>();
         int currentFrom = DEFAULT_FROM;
         int currentTo = MAX_RESULT_SIZE;
-        ParticipantRecordData rowData = new ParticipantRecordData(columnAliasEsPathMap);
-        Filterable filterable = FilterFactory.of(request);
-        //counting phase
         while (true) {
-            filterable.setFrom(currentFrom);
-            filterable.setTo(currentTo);
-            ParticipantWrapperResult filteredList = (ParticipantWrapperResult) filterable.filter(request.queryMap());
-            rowData.processData(filteredList, true);
-            if (filteredList.getTotalCount() < currentFrom) {
+            // For each batch of results, add the DTOs to the allResults list
+            filter.setFrom(currentFrom);
+            filter.setTo(currentTo);
+            ParticipantWrapperResult filteredSubset = (ParticipantWrapperResult) filter.filter(queryParamsMap);
+            allResults.addAll(filteredSubset.getParticipants());
+            // if the total count is less than the range we are currently on, stop fetching
+            if (filteredSubset.getTotalCount() < currentTo) {
                 break;
             }
             currentFrom = currentTo;
             currentTo += MAX_RESULT_SIZE;
         }
 
-        //data processing
-        currentFrom = DEFAULT_FROM;
-        currentTo = MAX_RESULT_SIZE;
-        ParticipantExcelGenerator generator = new ParticipantExcelGenerator();
-        generator.createHeader(rowData.getHeader());
-        int columnsNumber;
-        while (true) {
-            filterable.setFrom(currentFrom);
-            filterable.setTo(currentTo);
-            ParticipantWrapperResult filteredList = (ParticipantWrapperResult) filterable.filter(request.queryMap());
-            List<List<String>> participantRecords = rowData.processData(filteredList, false);
-            columnsNumber = getColumnsNumber(participantRecords);
-            if (filteredList.getTotalCount() < currentFrom) {
-                break;
-            }
-            generator.appendData(participantRecords);
-            currentFrom = currentTo;
-            currentTo += MAX_RESULT_SIZE;
-        }
-        generator.formatSizes(columnsNumber);
-        generator.writeInResponse(response);
+        return allResults;
+    }
 
-        return response.raw();
+    @Getter
+    @Setter
+    /** on retirement of feature-flag-export-new, this class should be promoted to DownloadParticipantListPayload */
+    private static class DownloadParticipantListPayloadNew {
```

Comment:
I don't think we need to call this "New" anymore if we rip out the other Payload class
Maybe use it in a separate class file, please